src/classes/RecipeRepository.js

```diff
@@ -6,10 +6,26 @@ class RecipeRepository {
   }
 
   static fromRecipeData(recipeData, ingredientsData) {
+    // Aggregate ingredients to remove duplicates.
+    recipeData.forEach((recipe) => {
+      const recipeIngredientMap = {};
+
+      recipe.ingredients.forEach((ingredient) => {
+        const recipeIngredient = recipeIngredientMap[ingredient.id];
+        if (!recipeIngredient) {
+          recipeIngredientMap[ingredient.id] = ingredient;
+        } else {
+          recipeIngredient.quantity.amount += ingredient.quantity.amount;
+        }
+      });
+
+      recipe.ingredients = Object.values(recipeIngredientMap);
+    });
+
```

Comment:
We were getting `422 errors` from the server because when we tried to post removing amounts we didn't have enough ingredient amounts. This was because there are duplicate ingredients in recipes and this was not factoring in duplicates. Now there are no longer duplicate ingredients in recipes.